src/compile/scale.ts

```diff
@@ -182,11 +174,8 @@ export function parseScaleComponent(model: Model): Dict<ScaleComponents> {
 
         // Add additional scales needed to support ordinal legends (list of values)
         // for color ramp.
-        if (channel === COLOR && model.legend(COLOR) && (fieldDef.type === ORDINAL || fieldDef.bin || fieldDef.timeUnit)) {
-          scales.colorLegend = parseColorLegendScale(model, fieldDef);
-          if (fieldDef.bin) {
-            scales.binColorLegend = parseBinColorLegendLabel(model, fieldDef);
-          }
+        if (channel === COLOR && model.legend(COLOR) && fieldDef.bin) {
```

Comment:
The comment above is outdated.  Now you only do it for bin. 

(why? -- add a comment why you need it for bin too, right now I don't understand why you still need it for bin.)